cql3/statements/select_statement.cc

```diff
@@ -1118,12 +1118,15 @@ query::partition_slice indexed_table_select_statement::get_partition_slice_for_g
     partition_slice_builder partition_slice_builder{*_view_schema};
 
     if (!_restrictions->has_partition_key_unrestricted_components()) {
+        // Computed token column to be added to index view restrictions
+        const column_definition& token_cdef = *_view_schema->clustering_key_columns().begin();
+        // Is secondary index with new token_column_computation type (computation name: token_v2)
+        const bool is_token_v2_index = token_cdef.type == long_type;
```

Comment:
In theory, we could support the old style too (and convert the blob token to a number and filter that). But I agree it's sufficient to support the new indexes.

Reply:
These calculations belong in the prepare state, not execution stage. Please move whatever parts are possible (those that don't consult `options` to prepare stage and store the results in members of indexed_table_select_statement. I don't expect any huge benefit from it, rather it reduces the amount of work we'll have to do later when we move even more stuff to the prepare stage.

Hm, I see you moved token_cdef from below, it's not a new variable. So no need to do anything, but keep in mind the preference to do stuff that doesn't depend on query parameters during preparation.